usaspending_api/awards/v2/views/idvs/funding_treemap.py

```diff
@@ -0,0 +1,96 @@
+from collections import OrderedDict
+from copy import deepcopy
+
+from psycopg2.sql import Identifier, Literal, SQL
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from usaspending_api.common.cache_decorator import cache_response
+from usaspending_api.common.helpers.sql_helpers import execute_sql_to_ordered_dictionary
+from usaspending_api.common.views import APIDocumentationView
+from usaspending_api.common.validator.award import get_internal_or_generated_award_id_model
+from usaspending_api.common.validator.tinyshield import TinyShield
+
+FUNDING_TREEMAP_SQL = SQL("""
+    with cte as (
+        select    award_id
+        from      parent_award
+        where     {award_id_column} = {award_id}
+        union all
+        select    cpa.award_id
+        from      parent_award ppa
+                  inner join parent_award cpa on cpa.parent_award_id = ppa.award_id
+        where     ppa.{award_id_column} = {award_id}
+    )
+    select
+        {columns}
+    from
+        cte
+        inner join awards pa on
+            pa.id = cte.award_id
+        inner join awards ca on
+            ca.parent_award_piid = pa.piid and
+            ca.fpds_parent_agency_id = pa.fpds_agency_id and
+            ca.type not like 'IDV\_%'
+        inner join financial_accounts_by_awards faba on
+            faba.award_id = ca.id
+        left outer join treasury_appropriation_account taa on
+            taa.treasury_account_identifier = faba.treasury_account_id {group_by};""")
+
+
+class IDVFundingBaseViewSet(APIDocumentationView):
+
+    """Returns File C funding totals associated with an IDV's children."""
+
+    @staticmethod
+    def _parse_and_validate_request(request: Request) -> dict:
```

Comment:
This should be `def _parse_and_validate_request(request: dict) -> dict:`... this was a late fix in the other endpoint.